lib/project_types/extension/commands/create.rb

```diff
@@ -22,22 +22,16 @@ def call(args, _)
             @ctx.abort(message_for_extension["create.errors.directory_exists", form.directory_name])
           end
 
-          return use_new_create_flow(form, message_for_extension) if supports_development_server?(form.type)
-
-          if form.type.create(form.directory_name, @ctx, getting_started: options.flags[:getting_started])
-            ExtensionProject.write_cli_file(context: @ctx, type: form.type.identifier)
-            ExtensionProject.write_env_file(
-              context: @ctx,
-              title: form.name,
-              api_key: form.app.api_key,
-              api_secret: form.app.secret
-            )
-
-            @ctx.puts(message_for_extension["create.ready_to_start", form.directory_name, form.name])
-            @ctx.puts(message_for_extension["create.learn_more", form.type.name])
-          else
-            @ctx.puts(message_for_extension["create.try_again"])
-          end
+          ShopifyCli::Result
+            .call { supports_development_server?(form.type.identifier) }
+            .then { |supported| create_extension(supported, form) }
+            .unwrap do |error|
+              if error.nil?
```

Comment:
Are we treating `nil` as success case? If `create_extension` returns `nil` but actually means to indicate success, it should return `ShopifyCli::Result.success(nil)` instead. The block passed to `unwrap` should only be concerned with error handling and should not deal with the happy path.